### DISCUSSION TITLE
[chocolatey] Native Image doesn't need Install-BinFile?

### DISCUSSION THREAD

**Discussion by mthmulders** [jreleaser/jreleaser#739] [Q&A] [answered]
I've recently submitted [a package](https://community.chocolatey.org/packages/mcs) (a native image) for review to Chocolatey. Today, one of the reviewers over there mentioned:

> Since the archive is extracted to a folder inside `$tool`s, there should not be a need to manually create a shim with `Install-BinFile`, as the shim would be created automatically.

I think this means we could remove the last line of the [chocolateyinstall.ps1.tpl](https://github.com/jreleaser/jreleaser/blob/main/core/jreleaser-templates/src/main/resources/META-INF/jreleaser/templates/native-image/chocolatey/tools/chocolateyinstall.ps1.tpl) for native-images?

**Comment by aalmiray** [comment, marked as answer]
Oh? Perfect! In this case templates for NATIVE_IMAGE and BINARY would need an update to remove the last line.

**Reply by TheCakeIsNaOH** [reply]
It also should mean the uninstall script is not needed.

**Reply by aalmiray** [reply]
@TheCakeIsNaOH the archive (such as app-1.2.3.zip) follows the following layout

```
app-1.2.3
├── LICENSE
├── README
└── bin
    └── app
```

As I understood it Chocolatey would figure things out if the archive is unpacked into `$tools`, even if the archive has a root entry and the executable is found inside a `bin` directory, right?

**Reply by TheCakeIsNaOH** [reply]
Correct. 

Chocolatey will search for any `.exe` files in the package installation directory (including in subdirectories), which is `$env:ChocolateyInstall\lib\<package>` and will create shims for them automatically.

**Reply by TheCakeIsNaOH** [reply]
`Install-Binfile` is useful for shimming `.bat` script, for shimming things outside of the package installation directory, or if the shim requires special options.

**Reply by aalmiray** [reply]
Perfect! Thank you for confirming, I can not safely edit the base templates to reduce unneeded parts 😄
